### PR TITLE
refactor(renterd): object routes

### PR DESCRIPTION
**Pull request by alexfreska**
- The single object and list APIs routes have been updated.

### CODE REVIEW

**Comment by alexfreska** [comment]
* **#749** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/749?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alexfreska and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: libs/renterd-types/src/bus.ts [hunk 1]
```diff
@@ -399,19 +398,19 @@ export type BucketDeleteParams = { name: string }
 export type BucketDeletePayload = void
 export type BucketDeleteResponse = void
 
-export type ObjectListParams = {
-  bucket?: string
+export type ObjectsParams = {
+  bucket: string
```
Comment (alexfreska) [inline libs/renterd-types/src/bus.ts:402]: is this now required or no?

Comment (ChrisSchinnerl) [reply libs/renterd-types/src/bus.ts:402]: It's not. But if you don't provide it you get results from all buckets. So the UI should make sure to provide it.

**Review by ChrisSchinnerl — APPROVED** [review]

**Comment by alexfreska** [comment]
### Merge activity

* **Sep 26, 7:45 AM EDT**: @alexfreska started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/SiaFoundation/web/749).

**Pull request merged**